### ISSUE TITLE
Tag failed stages/exit gracefully when inputs for later stages not found.

### ISSUE THREAD

**Issue by mdales** [quantifyearth/shark#44]
Now that we have more bits of shark working, I'm getting to the point where things in the container will fail, which is fine, but then we fail because there's no data for a later stage, and we blow up. That blow up looks a lot like a shark bug I've had to fight in the past, confusing me :)

When we can't find data for a later stage I need to stop markdown result generation and return what we have, as the output blocks would make it clear what happened from the stdout/stderr of the command. But we should also (incase things don't log) note when commands exit unsuccessfully, and add that to the output markdown.

Basically we've got shark to the point where it's not always what's wrong, but I need it to tell me the other ways in which I've messed up in the code I'm running in shark :)
